docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,208 @@
+# Satellite Service Separation
+
+## Table of Contents
+1. [Abstract](#abstract)
+2. [Background](#background): Current Satellite Design
+3. [Design](#design)
+4. [Rationale](#rationale)
+4. [Implementation](#implementation)
+5. [Open Issues](#open-issues)
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+### Current Satellite design
+
+Currently the Satellite is a single binary made of up the following services:
+
+#### overlay
+The overlay cache maintains critical information about nodes in the network (via the nodes table in Satellite.DB). One important use of overlay is to help select which storage nodes to upload files to.
```

Comment:
Let's use `overlay service` or just `overlay`. After adding actual caching to overlay, the naming ended up confusing, so it's better to start referring as `overlay`.